packages/twenty-front/src/modules/workflow/workflow-diagram/components/WorkflowRunDiagramCanvasEffect.tsx

```diff
@@ -24,25 +27,56 @@ export const WorkflowRunDiagramCanvasEffect = () => {
 
   const workflowId = useRecoilValue(workflowIdState);
 
-  const goBackToFirstWorkflowRunRightDrawerTabIfNeeded = useRecoilCallback(
+  const resetWorkflowRunRightDrawerTabIfNeeded = useRecoilCallback(
     ({ snapshot, set }) =>
-      () => {
+      ({
+        workflowSelectedNode,
+        stepExecutionStatus,
+      }: {
+        workflowSelectedNode: string;
+        stepExecutionStatus: WorkflowDiagramRunStatus;
+      }) => {
         const activeWorkflowRunRightDrawerTab = getSnapshotValue(
           snapshot,
           activeTabIdComponentState.atomFamily({
             instanceId: WORKFLOW_RUN_STEP_SIDE_PANEL_TAB_LIST_COMPONENT_ID,
           }),
         ) as WorkflowRunTabId | null;
 
+        const isInputTabDisabled = getIsInputTabDisabled({
+          stepExecutionStatus,
+          workflowSelectedNode,
+        });
+        const isOutputTabDisabled = getIsOutputTabDisabled({
+          stepExecutionStatus,
+        });
+
+        if (isNull(activeWorkflowRunRightDrawerTab)) {
+          const defaultTabId = isOutputTabDisabled
+            ? WorkflowRunTabId.NODE
+            : WorkflowRunTabId.OUTPUT;
+
+          set(
+            activeTabIdComponentState.atomFamily({
+              instanceId: WORKFLOW_RUN_STEP_SIDE_PANEL_TAB_LIST_COMPONENT_ID,
+            }),
+            defaultTabId,
+          );
+
+          return;
+        }
```

Comment:
The TabList component doesn't handle disabled tabs properly when it computes the initial tab. This code computes the initial tab.